pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1140,38 +1140,44 @@ type WorkflowStep struct {
 	// OnExit is a template reference which is invoked at the end of the
 	// template, irrespective of the success, failure, or error of the
 	// primary template.
-	// DEPRECATED: Use ExitTemplate.template instead.
+	// DEPRECATED: Use ExitHook.template instead.
 	OnExit string `json:"onExit,omitempty" protobuf:"bytes,11,opt,name=onExit"`
 
 	// Hooks holds exit hook which is invoked at the end of the
 	// template, irrespective of the success, failure, or error status of the primary template
-	Hooks *Hooks `json:"hooks,omitempty" protobuf:"bytes,12,opt,name=hooks"`
+	Hooks *LifecycleHook `json:"hooks,omitempty" protobuf:"bytes,12,opt,name=hooks"`
 }
 
-type Hooks struct {
+type LifecycleHook struct {
```

Comment:
sorry - I was not clear:

- `Hooks` -> `LifecyleHooks`
- `ExitTemplate` -> `LifecyleHook`

Reply:
make sence. will do it.